wsdl-core/src/main/java/io/ballerina/xsd/core/visitor/XSDVisitorImpl.java

```diff
@@ -0,0 +1,695 @@
+    public static final String ATTRIBUTE_ANNOTATION = "@xmldata:Attribute";
+    public static final String MEMBER_TYPES = "memberTypes";
+    public static final String UNION = "union";
+    public static final String VERTICAL_BAR = "|";
+    public static final String EMPTY_STRING = "";
+    public static final String RECORD_WITH_OPEN_BRACE = "record {|";
+    public static final String REQUIRED_FIELD_NOT_FOUND_ERROR = "Required field is not found in <complexType>: '%s'";
+    public static final String ONE = "1";
+    public static final String XMLDATA_CHOICE = "@xmldata:Choice";
+    public static final String CHOICE_NAME = "ChoiceOption";
+    public static final String EMPTY_ARRAY = "[]";
+    public static final String XMLDATA_SEQUENCE = "@xmldata:Sequence";
+    public static final String SEQUENCE_NAME = "SequenceGroup";
+    public static final String XMLDATA_ORDER = "@xmldata:SequenceOrder";
+    public static final String RESTRICTION = "restriction";
+    public static final String LIST = "list";
+    public static final String ITEM_TYPE = "itemType";
+
+    private final ArrayList<String> imports = new ArrayList<>();
+    private final Map<String, String> extensions = new LinkedHashMap<>();
+    private final Map<String, String> rootElements = new LinkedHashMap<>();
+    private final Map<String, String> nameResolvers = new LinkedHashMap<>();
+    private final Map<String, String> nestedElements = new LinkedHashMap<>();
+    private final Map<String, ArrayList<String>> enumerationElements = new LinkedHashMap<>();
+    private String contentField;
+    private String targetNamespace;
+
+    public XSDVisitorImpl(String contentField) {
+        this.contentField = contentField;
+    }
+
+    public String getContentField() {
+        return contentField;
+    }
+
+    public void setContentField(String contentField) {
+        this.contentField = contentField;
+    }
+
+    @Override
+    public String visit(Element element) throws Exception {
+        if (element.isSubType()) {
+            return this.visit(element, true);
+        }
+        Node node = element.getNode();
+        StringBuilder builder = new StringBuilder();
+        builder.append(addNamespace(this, getTargetNamespace()));
+        builder.append(PUBLIC).append(WHITESPACE).append(TYPE).append(WHITESPACE);
+        if (!node.hasChildNodes()) {
+            return handleSingleElementNode(node, builder);
+        }
+        return handleElementsWithChildNodes(node, builder);
+    }
+
+    @Override
+    public String visit(Element element, boolean subElement) throws Exception {
+        Node node = element.getNode();
+        StringBuilder builder = new StringBuilder();
+        Node nameNode = node.getAttributes().getNamedItem(NAME);
+        Node typeNode = node.getAttributes().getNamedItem(TYPE);
+        try {
+            typeNode = visitNestedElements(node, nameNode, typeNode);
+        } catch (Exception e) {
+            throw new Exception(e);
+        }
```

Comment:
Any reason to re throwing the error here? We can remove the try, catch block if there aren't any.

Reply:
Fixed in [a3cf525](https://github.com/ballerina-platform/wsdl-tools/pull/17/commits/a3cf52556c04e2dba2d0ac446a79348a61d012d6)